File: rezerveTicketing/ticketing/src/main/java/com/endava/demo/dto/TicketDTO.java
package com.endava.demo.dto;

import com.endava.demo.model.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class TicketDTO {


        private int id;
        private LocalDate date;
        private int seatNumber;
        private int row;
        private Movie movie;
        private Concert concert;
        private Sport sport;
        private TicketType ticketType;
        private int quantity;
        private String selectedTime;
        private List<TicketSeatDTO> ticketSeats;

    }

File: ticketing-client/src/DTOs/TicketDTO.ts
import { Movie } from "../UI/MovieComp/Movie";
import { Concert } from "../UI/ConcertComp/Concert";
import { TicketType } from "../UI/CartComp/TicketType";
import { Seat } from "../interfaces/Seat";
import { Sport } from "../UI/SportComp/Sport";

export interface TicketDTO {
  id: number;

  // needs modification for every entity added to cart
  movie: Movie;
  concert: Concert;
  sport: Sport;

  date: Date;
  row: number;
  seatNumber: number;
  ticketType: TicketType;
  ticketSeats: Seat[];
  quantity: number;
  selectedTime: string;
}
